dd-sdk-android/src/main/kotlin/com/datadog/android/v2/core/internal/net/DataOkHttpUploader.kt

```diff
@@ -35,7 +36,12 @@ internal class DataOkHttpUploader(
         batch: List<ByteArray>,
         batchMeta: ByteArray?
     ): UploadStatus {
-        val request = requestFactory.create(context, batch, batchMeta)
+        val request = try {
+            requestFactory.create(context, batch, batchMeta)
+        } catch (e: Exception) {
+            internalLogger.errorWithTelemetry("Unable to create the upload request.", e)
+            return UploadStatus.UNKNOWN_ERROR
```

Comment:
should we create a dedicated error for this case ?

Reply:
Yes, I think we should. Also we should show the exception happened to the user as well, so let's log it with the target=user, not contributor.